tests/components/homekit/test_type_cameras.py

```diff
@@ -0,0 +1,352 @@
+        return_value=_get_working_mock_ffmpeg(),
+    ):
+        acc.set_selected_stream_configuration(
+            "ARUCAQEBEDMD1QMXzEaatnKSQ2pxovYCNAEBAAIJAQECAgECAwEAAwsBAgAFAgLQAgMBHgQXAQFjAgQ768/RAwIrAQQEAAAAPwUCYgUDLAEBAwIMAQEBAgEAAwECBAEUAxYBAW4CBCzq28sDAhgABAQAAKBABgENBAEA"
+        )
+        await acc.stop_stream(session_info)
+        await hass.async_block_till_done()
+
+
+async def test_camera_stream_source_fails(hass, run_driver, events):
+    """Test a camera that can stream and we cannot get the source from the entity."""
+    await async_setup_component(hass, ffmpeg.DOMAIN, {ffmpeg.DOMAIN: {}})
+    await async_setup_component(
+        hass, camera.DOMAIN, {camera.DOMAIN: {"platform": "demo"}}
+    )
+
+    entity_id = "camera.demo_camera"
+
+    hass.states.async_set(entity_id, None)
+    await hass.async_block_till_done()
+    acc = Camera(hass, run_driver, "Camera", entity_id, 2, {},)
+    await acc.run_handler()
+
+    assert acc.aid == 2
+    assert acc.category == 17  # Camera
+
+    stream_service = acc.get_service("CameraRTPStreamManagement")
+    endpoints_config_char = stream_service.get_characteristic("SetupEndpoints")
+    assert endpoints_config_char.setter_callback
+    stream_config_char = stream_service.get_characteristic(
+        "SelectedRTPStreamConfiguration"
+    )
+    assert stream_config_char.setter_callback
+    acc.set_endpoints(
+        "ARAzA9UDF8xGmrZykkNqcaL2AgEAAxoBAQACDTE5Mi4xNjguMjA4LjUDAi7IBAKkxwQlAQEAAhDN0+Y0tZ4jzoO0ske9UsjpAw6D76oVXnoi7DbawIG4CwUlAQEAAhCyGcROB8P7vFRDzNF2xrK1Aw6NdcLugju9yCfkWVSaVAYEDoAsAAcEpxV8AA=="
+    )
+
+    session_info = {
+        "id": "mock",
+        "v_srtp_key": "key",
+        "a_srtp_key": "key",
+        "v_port": "0",
+        "a_port": "0",
+        "address": "0.0.0.0",
+    }
+    acc.sessions[UUID("3303d503-17cc-469a-b672-92436a71a2f6")] = session_info
+
+    with patch(
+        "homeassistant.components.demo.camera.DemoCamera.stream_source",
+        side_effect=OSError,
+    ), patch(
+        "homeassistant.components.homekit.type_cameras.HAFFmpeg",
+        return_value=_get_working_mock_ffmpeg(),
+    ):
+        acc.set_selected_stream_configuration(
+            "ARUCAQEBEDMD1QMXzEaatnKSQ2pxovYCNAEBAAIJAQECAgECAwEAAwsBAgAFAgLQAgMBHgQXAQFjAgQ768/RAwIrAQQEAAAAPwUCYgUDLAEBAwIMAQEBAgEAAwECBAEUAxYBAW4CBCzq28sDAhgABAQAAKBABgENBAEA"
+        )
+        await acc.stop_stream(session_info)
+        await hass.async_block_till_done()
+
+
+async def test_camera_with_no_stream(hass, run_driver, events):
+    """Test a camera that cannot stream."""
+    await async_setup_component(hass, ffmpeg.DOMAIN, {ffmpeg.DOMAIN: {}})
+    await async_setup_component(hass, camera.DOMAIN, {camera.DOMAIN: {}})
+
+    entity_id = "camera.demo_camera"
+
+    hass.states.async_set(entity_id, None)
+    await hass.async_block_till_done()
+    acc = Camera(hass, run_driver, "Camera", entity_id, 2, {},)
+    await acc.run_handler()
+
+    assert acc.aid == 2
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5462445)